documents/converters.go

```diff
@@ -195,7 +200,13 @@ func toProtocolAttributes(attrs map[AttrKey]Attribute) (pattrs []*coredocumentpb
 		case AttrBytes:
 			pattr.Value = &coredocumentpb.Attribute_ByteVal{ByteVal: attr.Value.Bytes}
 		case AttrTimestamp:
-			pattr.Value = &coredocumentpb.Attribute_TimeVal{TimeVal: attr.Value.Timestamp}
+			buf := new(bytes.Buffer)
+			err := binary.Write(buf, binary.BigEndian, attr.Value.Timestamp.Seconds)
```

Comment:
We are ignoring the nanoseconds here. Lets embed nanoseconds and increase the `maxTimeByteLength` to 12?

Reply:
```suggestion
			binary.BigEndian.PutUint64(buf[:], attr.Value.Timestamp.Seconds)
```

Reply:
seconds are int64

Reply:
We intentionally left the nanoseconds out, since blockchains are not that accurate anyways. If you see a big problem, I am open to change it to 12.

Reply:
> is that simple casting var y uint64 = uint64(x)
Yees, it is a direct conversion. 

> We intentionally left the nanoseconds out, since blockchains are not that accurate anyways.
The attribute is also given by the user. ignoring nano will change the client data. Hence, I feel we should add nano as well.

Reply:
Actually, I agree there. I will add nanoseconds as well. On the other hand I rather keep using `binary.Write` and `binary.Read` since it is more safe in case the user input exceeds the internal representation and error out instead of `PutUintx` panicking.

Reply:
but mostly it is easier to deal with buffers.

Reply:
@vedhavyas I disagree, I think rounding to 0 nanoseconds is fine. we can always have a second type that is more accurate later.

Reply:
@lucasvo Why is losing precision on nano when user provided the timestamp in nano? I feel we shouldn't change the user-provided data here

Reply:
> but mostly it is easier to deal with buffers, since in this case we already know the size of the buffer as we only deal with int64/int32
But this wont panic if we have a fixed byte array instead of slice/buffer. But upto you though

Reply:
@lucasvo if nano is not provided by the user when defining the data parameter `2006-01-02T15:04:05Z07:00` then nanos will be zero. If they decide to add the nano precision: `2006-01-02T15:04:05.999999999Z07:00` then it will be part of the byte array serialized. I think is a free option for us to keep the nanos as well.